File: InvestmentTrackerAPI/InvestmentTrackerAPI/Requests/Transaction/TransactionRequest.cs
using System.ComponentModel.DataAnnotations;

namespace InvestmentTrackerAPI.Requests.Transactions
{
    public class TransactionRequest
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        public DateTime Date { get; set; } = DateTime.UtcNow;

        public decimal Amount { get; set; }

        public int CategoryId { get; set; }

        public string? Description { get; set; }

        public int TypeId { get; set; }

    }
}

File: InvestmentTrackerUI/shared/types/TransactionRequest.ts
export class TransactionRequest {
    typeId? : string
    categoryId? : string
    amount: number
    description? : string
    date? : Date

    constructor(data: Partial<TransactionRequest>) {
    this.typeId = data.typeId
    this.categoryId = data.categoryId
    this.amount = data.amount ?? 0
    this.description = data.description
    this.date = data.date
    
  }

  toJson(): Record<string, any> {
    return {
      typeId: this.typeId,
      categoryId: this.categoryId,
      amount: this.amount,
      description: this.description,
      date: this.date,
    }
  }

}